frontend/src/metabase/dashboard/components/ClickBehaviorSidebar/ClickBehaviorSidebarMainView/ClickBehaviorSidebarMainView.tsx

```diff
@@ -91,7 +88,7 @@ function ClickBehaviorSidebarMainView({
 
   return (
     <>
-      {!dashcard.action && (
+      {!isActionDashboardCard(dashcard) && (
```

Comment:
I'm a little confused when I see how this variable & function are typed. You can see that the `dashCard` is always `DashboardOrderedCard`, so I'd expect `!isActionDashboardCard(dashcard)` to always be true and we might never need to have this check at all. What do you think?
![image](https://user-images.githubusercontent.com/1937582/200283421-d32a74e0-07bf-4015-bbf0-6db80d9ee6b7.png)

Reply:
I'm a little confused here. Part of the diff is adding ` | ActionDashboardCard ` to the types for this function's props because sometimes this actually is an ActionDashboardCard

Reply:
Ok, my point is that the `dashcard` here can only be `DashboardOrderedCard` since its type is declared here
https://github.com/metabase/metabase/blob/19cf41bd37f431a4e3fce400235bffc65c466295/frontend/src/metabase/dashboard/components/ClickBehaviorSidebar/ClickBehaviorSidebarMainView/ClickBehaviorSidebarMainView.tsx#L67

So, that sounds a bit weird saying that we know `dashcard` prop in `ClickBehaviorSidebarMainView` component is always a `DashboardOrderedCard` but we still need to check if it isn't an `ActionDashboardCard` (since if it's always a `DashboardOrderedCard` this clause `!isActionDashboardCard(dashcard)` will always return true).

It'd make sense for this check to exist if we say oh the `dashcard` passed to this `ClickBehaviorSidebarMainView` component could either be an `ActionDashboardCard` or something else.

Hope that is a bit clearer.

Reply:
Oh I understand now. `ActionDashboardCard` and `DashboardOrderedCard` are not mutually exclusive, so all `ActionDashboardCard`s are also `DashboardOrderedCard`s, but not all all `DashboardOrderedCard`s are `ActionDashboardCard`s.

That's why we need this check, so we can determine if it's part of the subset that needs this special handling. This type relationship is also why we need a type guard instead of a simple property check - typescipt throws a fit if you try to check the `action` property in this component because not all `DashboardOrderedCard`s have it, even as an optional property.